src/components/molecules/PluginEditor/hooks.ts

```diff
@@ -37,138 +33,187 @@ const defaultPosition = {
-    },
-    {
-      RightTop: { section: "right", area: "top" },
-      RightMiddle: { section: "right", area: "middle" },
-      RightBottom: { section: "right", area: "bottom" },
-    },
-  ];
-
-  const handleFileDownload = (body?: string, filename?: string) => {
+  const handleDownload = (body?: string, filename?: string) => {
     if (!body) return;
     fileDownload(body, filename ?? "untitled.js");
   };
 
-  const handleAlignSystemUpdate = (widget: Widget, newLoc: Position) => {
-    setAlignSystem(() => {
-      const alignment =
-        newLoc.section === "center" || newLoc.area === "middle" ? "centered" : "start";
-      const newSection: WidgetSection = {
-        top: {
-          widgets: newLoc.area === "top" ? [widget] : undefined,
-          align: alignment,
-        },
-        middle: {
-          widgets: newLoc.area === "middle" ? [widget] : undefined,
-          align: alignment,
-        },
-        bottom: {
-          widgets: newLoc.area === "bottom" ? [widget] : undefined,
-          align: alignment,
-        },
-      };
-
-      const newZone: WidgetZone = {
-        left: {
-          ...(newLoc.section === "left" ? newSection : undefined),
-        },
-        center: {
-          ...(newLoc.section === "center" ? newSection : undefined),
-        },
-        right: {
-          ...(newLoc.section === "right" ? newSection : undefined),
-        },
-      };
-      setCurrentPosition(newLoc);
-      return { outer: newZone };
-    });
-  };
-  const handleAlignSystemToggle = () => {
-    setShowAlignSystem(!showAlignSystem);
-  };
+  const handleAlignSystemToggle = useCallback(() => {
+    setShowAlignSystem(s => !s);
+  }, []);
 
   const handleInfoboxToggle = () => {
     setShowInfobox(!showInfobox);
   };
 
-  const handleFileOpen = async (e: ChangeEvent<HTMLInputElement>) => {
-    e.preventDefault();
-    const file = e.currentTarget.files?.[0];
+  const handleOpen = useFileInput(files => {
+    const file = files?.[0];
     if (!file) return;
     const reader = new FileReader();
 
     reader.onload = async e2 => {
       const body = e2?.target?.result;
       if (typeof body != "string") return;
-      setSourceCode({ fileName: file.name, body });
-      setHardSourceCode({ fileName: file.name, body });
+      setFileName(file.name);
+      setSourceCode(body);
     };
     reader.readAsText(file);
-  };
-
-  const widget = useMemo(() => {
-    return {
-      id: "xxx",
-      // extended: true,
-      __REEARTH_SOURCECODE: sourceCode.body,
-    };
-  }, [sourceCode.body]);
+  });
 
-  const handleFileReset = useCallback(() => {
+  const handleReset = useCallback(() => {
     if (confirm("Are you sure you want to reset?")) {
-      setSourceCode(hardSourceCode);
-      handleAlignSystemUpdate(widget, defaultPosition);
+      setSourceCode(originalSourceCode);
+      setFileName(originalFileName);
+      setCurrentPosition(defaultPosition);
+      handleRun();
```

Comment:
If I reset once, the WAS and other settings reset, but not the originalSourceCode. I have to do a second reset for it to happen.

Reply:
`originalSoruceCode` and `originalFileName` are constants and cannot be changed.

Reply:
Yep I know they are constants. What I mean is it doesn't display the originalSourceCode on first Reset. I have to reset a second time for it to show

Reply:
fixed: [`d674fe9` (#274)](https://github.com/reearth/reearth-web/pull/274/commits/d674fe90aa409db489ade67f7e14fb7f8b1d5873)